qdl.h

```diff
@@ -48,6 +49,11 @@ struct qdl_device_usb
 	size_t out_chunk_size;
 };
 
+struct qdl_device_sim
+{
+	struct qdl_device base;
+};
```

Comment:
I think qdl_device_foo should be internal to foo.h

Reply:
tbh I don't want to introduce a new header file just for one struct definition. If there are more things to move to a separate header in the future, that will definitely make sense.

Reply:
Then just sim.c, if that's the only place where it is used.